book/src/zcash/ywallet-demo.md

```diff
@@ -128,36 +131,51 @@ run):
 RUST_LOG=debug cargo run --bin server
 ```
 
-In the second one, the Coordinator, run (in the same folder where key generation
-was run):
+### Registering users
+
+In order to interact with the server, you will need to register users. For this
+guide we will need two. In a new terminal, run the following command for user
+"alice" (replace the password if you want):
 
 ```
-cargo run --bin coordinator -- -C redpallas --http -r -
+curl --data-binary '{"username": "alice", "password": "foobar10", "pubkey": ""}' -H 'Content-Type: application/json' http://127.0.0.1:2744/register
 ```
 
-And then:
+It will output "null". Also register user "bob":
 
-- It should read the public key package from `public-key-package.json`.
-- Type `2` for the number of participants.
-- Copy the session ID that is printed.
+```
+curl --data-binary '{"username": "bob", "password": "foobar10", "pubkey": ""}' -H 'Content-Type: application/json' http://127.0.0.1:2744/register
+```
 
+You only need to do this once, in case you want to sign more than one
+transaction. If for some reason you want to start over, close the server and
+delete the `db.sqlite` file.
 
-In the third terminal, Participant 1, run the following, replacing
-`<SESSION_ID>` with the session ID you have just copied:
+Feel free to close this terminal, or reuse it for the next step.
 
+```admonish warning
+Do not use passwords that you use in practice; use dummy ones instead. (You
+shouldn't reuse passwords anyway!) For real world usage you would need to take
+more care to not end up writing the password to your shell history. (In real
+world usage we'd expect this to be done by applications anyway.)
 ```
-cargo run --bin participant -- -C redpallas --http --key-package key-package-1.json -s <SESSION_ID>
-```
 
-In the fourth terminal, for Participant 2, run the following, again replacing
-the session ID:
+### Coordinator
+
+In the second terminal, the Coordinator, run (in the same folder where key
+generation was run):
 
 ```
-cargo run --bin participant -- -C redpallas --http --key-package key-package-2.json -s <SESSION_ID>
+export PW=foobar10
+cargo run --bin coordinator -- -C redpallas --http -u alice -w PW -S alice,bob -r -
 ```
 
-Go back to the Coordinator CLI:
+We will use "alice" as the Coordinator, so change the value next to `export PW=`
+if you used another password when registering "alice".
 
+And then:
+
+- It should read the public key package from `public-key-package.json`.
 - When prompted for the message to be signed, paste the SIGHASH printed by the
```

Comment:
Here it's waiting for participants to send their commitments first, maybe we should add this before mentioning the prompt for the message to be signed

Reply:
Ah I see, I ended up using https://github.com/ZcashFoundation/frost-zcash-demo/pull/265 to write these docs, and the flow changed there. Since that's already approved I think it's OK, we can merge the docs after both that and the user registration PR merge.